snapcraft/commands/daemon.start

```diff
@@ -619,7 +619,7 @@ fi
 
 # LXD
 ## Check for existing LXDs
-for pid in $(pgrep -f "lxd(\.debug)? --logfile"); do
+for pid in $(pgrep --euid 0 --full "lxd(\.debug)? --logfile"); do
```

Comment:
in the example in your commit, why is server_pid 1293, but this command doesn't match that process?

Reply:
Also why not --uid?

Reply:
The `server_pid 1293` is relative to the container's PID NS which is why it doesn't match the host's view.

Reply:
As to why not `--uid` instead of `--euid` it shouldn't really matter here as they should both be `0`. The distinction becomes relevant when using setuid binaries.

Reply:
Also, in other scripts we use `id -u` to ensure the EUID is `0` so this should be more consistent.